### DISCUSSION TITLE
Required parameter missing or invalid: items

### DISCUSSION THREAD

**Discussion by yergneb** [EvgeniyMukhamedjanov/liquid-ajax-cart#89] [Q&A]
The hidden input within the cart has the value of the product ID, not the variant ID resulting in a 400 Bad Request error. Any help here?

**Comment by EvgeniyMukhamedjanov** [comment]
Are talking about product form? Or how do you send the request and why you use product ID instead of variant ID?

**Reply by yergneb** [reply]
Apologies for not being very clear. Was after a long day!

Yes, talking about the product form. By default, it adds a hidden input with the product ID, not the variant ID.

```
<ajax-cart-product-form>
  {% form 'product', product %}
    <!-- form content -->
  
    <button type="submit" name="add">
      Add to cart
    </button>
  {% endform %}
</ajax-cart-product-form>
```

I've got it working by using liquidAjaxCart.add() and passing the variant ID through, but was just wondering if this issue with the form is an issue or something I'm missing?

**Reply by arkodeepg** [reply]
can you once try adding the variant id, I had that issue and had to put the variant id in place
```
<ajax-cart-product-form>
  {% form 'product', product %}
    <!-- form content -->
    <input type="hidden" name="id"
 value="{{ product.selected_or_first_available_variant.id }}">
  <input type="text" name="quantity" min="1" value="1">
    <button type="submit" name="add">
      Add to cart
    </button>
  
    <div data-ajax-cart-errors="form"></div>
  {% endform %}
</ajax-cart-product-form>
```

**Reply by EvgeniyMukhamedjanov** [reply]
@yergneb 
Yeah, the variant id should be there.
The error message you saw came from Shopify, not from Liquid Ajax Cart.
Shopify requires each product form to have variant id so that Shopify knows which variant should be added to the cart (even if the product has only one default variant):
https://shopify.dev/docs/themes/architecture/templates/product#the-product-form

To prove that you can remove liquid ajax cart and submit the form without the variant id. The product won't be added to the cart.
